### PR TITLE
Resolution to number conversion for python methods

### PR DESCRIPTION
**Pull request by Jieran-S**
Implemented resolution to n_clust conversion via binary search for Python-based methods. 

Methods modified: scanpy, STAGATE, Spaceflow

Tested in the workflow pipeline. Should all work fine.

### CODE REVIEW

File: method/SpaceFlow/method_spaceflow.py [hunk 1]
```diff
@@ -155,12 +166,15 @@ def get_anndata(args):
     nn = 15
 
 # Raise a warning that clustering is based on resolution and not n_clusters
-warnings.warn("The `n_clusters` parameter was not used; config['res'] used instead.")
+# warnings.warn("The `n_clusters` parameter was not used; config['res'] used instead.")
 
 # Segment the domains given the resolution
-sf.segmentation(domain_label_save_filepath=label_file, n_neighbors=nn, resolution=res)
```
Comment (niklasmueboe) [inline method/SpaceFlow/method_spaceflow.py:161]: Does this function use Leiden clustering?

Comment (Jieran-S) [reply method/SpaceFlow/method_spaceflow.py:161]: Yes. Check here:

https://github.com/hongleir/SpaceFlow/blob/58d1ab03f1538791f710db840d36a08f396491fb/SpaceFlow/SpaceFlow.py#L289-L314

File: method/STAGATE/method_STAGATE.py [hunk 2]
```diff
@@ -78,13 +78,22 @@
...
 tf.compat.v1.disable_eager_execution()
 # the location of R (used for the mclust clustering)
 import scipy as sp
 
+# import res-n_clust tuning function
+import sys
+from pathlib import Path
+
+# Add the parent directory of the current file to sys.path
+method_dir = Path(__file__).resolve().parent.parent  # Navigate two levels up
```
[the remark this answers is not in the document]
Comment (niklasmueboe) [reply method/STAGATE/method_STAGATE.py:92]: > Should we create an issue for this matter so we can merge the PR as a temporary working solution?

That sounds like a good idea. Can you create an issue and also link it to this PR.
After that I'd say we can merge it.

[the remark this answers is not in the document]
Comment (Jieran-S) [reply method/STAGATE/method_STAGATE.py:92]: Issue created with number 234. This PR can be merged now.

**Review by niklasmueboe — APPROVED** [review]

**Pull request merged**